File: Monopoly/src/monopoly/Property.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package monopoly;

/**
 *
 * @author nikandrosmavroudakis
 */
public class Property extends Card {
    
    private Group group;
    private Player owner;
    private int numberOfHouses;
    
    public Property(String text, Group group) {
        super(text);
        this.group = group;
        numberOfHouses = 0;
        owner = null;
    }
    
    public void build() {
        if(numberOfHouses < 5) {
            numberOfHouses++;
        }
    }
    
    public void buyIt(Player owner) {
        this.owner = owner;
    }
    
    public Player getOwner() {
        return owner;
    }
    
    public int getRent() {
        // "SELECT rent" + numberOfHouses + " FROM properties WHERE name=text"
        
        return 200;
    }
}

File: House/Assets/Scripts/Objects/Property.cs
public class Property : Card {

	private string group;
	private Player owner;
	private int numberOfHouses;

	public Property(string text, string group) : base(text) {
		this.group = group;
		numberOfHouses = 0;
		owner = null;
	}

	public void BuyIt(Player owner) {
		this.owner = owner;
	}

	public void Build() {
		if(numberOfHouses < 5) {
			numberOfHouses++;
		}
	}

	public string GetGroup() {
		return group;
	}

	public Player GetOwner() {
		return owner;
	}

	public int GetNumberOfHouses() {
		return numberOfHouses;
	}
}
